integreat_cms/cms/views/poi_categories/poi_category_form_view.py

```diff
@@ -60,6 +65,29 @@ def get_permission_required(self):
         # If the form is just retrieved, require the view permission
         return ("cms.view_poicategory",)
 
+    def get_poi_category_form(self):
+        """
+        Retrieve and instantiate the POICategoryForm
+
+        :returns: poi_category_form
+        :rtype: ~django.forms.formsets.POICategoryForm
+        """
+        if self.poi_category_form:
+            return self.poi_category_form
+
+        if self.object:
+            poi_category_instance = POICategory.objects.filter(
+                id=self.object.id
+            ).first()
+            return POICategoryForm(
+                data=self.request.POST or None, instance=poi_category_instance
+            )
+
+        poi_category_instance = POICategory()
+        return POICategoryForm(
+            data=self.request.POST or None, instance=poi_category_instance
+        )
+
```

Comment:
```suggestion
```
We don't need this custom method, the Django form view already handles this for us.
Just set the correct [`form_class`](https://docs.djangoproject.com/en/4.1/ref/class-based-views/mixins-editing/#django.views.generic.edit.FormMixin.form_class) attribute of the view, then this class will be used when instantiating the form.

You also have to remove
```
   #: The fields of this :class:`~django.views.generic.edit.ModelFormMixin`
   #: (see :attr:`~django.views.generic.edit.ModelFormMixin.fields`)
   fields = ["id"]
```

Now because the `fields` setting is now contained in the form class and cannot be given in the view again.